pkg/envoy/rds.go

```diff
@@ -40,6 +43,20 @@ func (l RouteDomainCheck) Run() outcomes.Outcome {
 	foundAnyRouteDomains := false
 	foundSpecificRouteDomain := false
 
+	var possibleDomains []string
+	if len(l.Domain) > 0 {
+		possibleDomains = append(possibleDomains, l.Domain)
+	}
+	if l.Client != nil {
+		svcs, err := kuberneteshelper.GetMatchingServices(l.Client, l.Labels, l.Namespace)
+		if err != nil {
+			return outcomes.FailedOutcome{Error: err}
+		}
+		for _, svc := range svcs {
+			possibleDomains = append(possibleDomains, svc.Name+"."+svc.Namespace)
+		}
+	}
+
```

Comment:
If there are no `possibleDomains` could the check be terminated early with an `Info` outcome?

Reply:
Good catch, that should work. I wonder if we could go as far as throwing an error in that case. I can't think of a successful scenario where `possibleDomains` would end up empty.